packages/libs/hd-wallet/src/index.ts

```diff
@@ -1 +1,253 @@
+ * @returns {([string, string] | [string, string][])} - Depending on the input, either a tuple for a single key pair or an array of tuples for a range of key pairs, with the private key encrypted if a password is provided.
+ * @throws {Error} Throws an error if the seed buffer is not provided, if the indices are invalid, or if encryption fails.
+ */
+export function kadenaGenKeypair(
+  seedBuffer: Uint8Array,
+  indexOrRange: number | [number, number],
+  password?: string,
+): [string, string] | [string, string][] {
+  if (seedBuffer === undefined) throw new Error('No seed provided.');
+
+  if (Array.isArray(indexOrRange)) {
+    const [start, end] = indexOrRange;
+    if (start > end)
+      throw new Error(
+        'Invalid range: start index must be less than or equal to end index.',
+      );
+
+    const keyPairs: [string, string][] = [];
+    for (let i = start; i <= end; i++) {
+      const { publicKey, privateKey } = deriveKeyPair(seedBuffer, i);
+      const privatekeyToUse =
+        password !== undefined
+          ? encryptPrivateKey(Buffer.from(privateKey, 'hex'), password)
+          : privateKey;
+      keyPairs.push([publicKey, privatekeyToUse]);
+    }
+    return keyPairs;
+  } else {
+    const { publicKey, privateKey } = deriveKeyPair(seedBuffer, indexOrRange);
+
+    const privatekeyToUse =
+      password !== undefined
+        ? encryptPrivateKey(Buffer.from(privateKey, 'hex'), password)
+        : privateKey;
+    return [publicKey, privatekeyToUse];
+  }
+}
+
+/**
+ * Decrypts an encrypted private key using the provided password.
+ * This function is a wrapper for the internal decryption logic, intended
+ * for public-facing API usage where the private key encryption follows
+ *
+ * @param {string} encryptedPrivateKey - The encrypted private key as a Base64 encoded string.
+ * @param {string} password - The password used to encrypt the private key.
+ * @returns {Uint8Array} The decrypted private key.
+ * @throws {Error} Throws an error if decryption fails.
+ */
+export function kadenaDecryptPrivateKey(
+  encryptedPrivateKey: string,
+  password: string,
+): Uint8Array {
+  try {
+    return decryptPrivateKey(encryptedPrivateKey, password);
+  } catch (error) {
+    console.error('Failed to decrypt the private key:', error);
+    throw new Error(
+      'Decryption failed. The provided password may be incorrect, or the encrypted key is corrupted.',
+    );
+  }
+}
+
+/**
+ * Generates a public key from a seed buffer and an index.
+ *
+ * @param {Uint8Array} seedBuffer - The seed buffer used for key generation.
+ * @param {number} [index] - The optional index of the public key to retrieve, defaults to 0.
+ * @returns {string} - The public key in hexadecimal string format.
+ * @throws {Error} Throws an error if the seed buffer is not provided or if the index is invalid.
+ */
+export function kadenaGetPublic(
+  seedBuffer: Uint8Array,
+  index: number = 0,
+): string {
```

Comment:
To be similar to legacy interface

```suggestion
export declare const kadenaGetPublicFromSeed: (
  password: string,
  seed: string | Uint8Array,
  derivationPath: string // default value  `m/44'/626'/{$index}'/0/0`
) => Uint8Array;

export declare const kadenaGetPublicFromSeed: (
  password: string,
  seed: string | Uint8Array,
  ...derivationPath: string[]
) => Uint8Array[];
```